fault_injection/fi_ibex.py

```diff
@@ -178,11 +178,16 @@ def main(argv=None):
     # Save metadata.
     metadata = {}
     metadata["datetime"] = datetime.now().strftime("%m/%d/%Y, %H:%M:%S")
-    metadata["cfg"] = cfg
-    metadata["fpga_bitstream"] = cfg["target"]["fpga_bitstream"]
-    # TODO: Store binary into database instead of binary path.
-    # (Issue lowrisc/ot-sca#214)
-    metadata["fw_bin"] = cfg["target"]["fw_bin"]
+    # Store bitstream information.
+    metadata["fpga_bitstream_path"] = cfg["target"]["fpga_bitstream"]
+    metadata["fpga_bitstream_crc"] = helpers.file_crc(cfg["target"]["fpga_bitstream"])
```

Comment:
I like the idea of the CRC!

What I wondering is if it is of any use if we're not storing the binary/bitstream? Or do you envision a use case where one uses e.g. the bistream currently in the repo but experiments with different binaries? And then when going back later one can use the bistream that was in the repo back then?

I think this could work, because in most cases one will have to go back in time in the repo anyway to reproduce results (to do that the Git Hash would be nice! I don't know if we're currently doing that).

But just from a perspective of saving storage, I don't think it's worth it. Bitstream and binary together are maybe 100 MB at most, but the traces themselves are quickly in the GB domain. Or what is your view?

Reply:
I implemented this because I know I will make the following in the future:
Because I want to save some storage space for small experiments, I'll omit the `save binary` argument.
When then debugging an issue (with the bitstream or the binary), the CRC field in the database then could be handy as I simply need to compare the output of `crc32 <bitstream/binary>` with the entry.

Git hash is a good idea, I'll quickly implement this.

Reply:
Keeping the CRC always makes definitely sense and I see the value for debugging. I am just wondering if we should avoid the possibility of getting into the situation where "debugging" is needed by always storing the binary. But again, when using the ES silicon with the firmware in the chip already, there is no point int storing the binary. So, I agree to keep storing the binaries optional.

Thanks for adding the hash! :+1: